astropy/wcs/wcs.py

```diff
@@ -187,14 +187,21 @@ class NoConvergence(Exception):
 
     """
 
-    def __init__(self, *args, **kwargs):
+    def __init__(self, *args, best_solution=None, accuracy=None, niter=None,
+                 divergent=None, slow_conv=None, **kwargs):
         super().__init__(*args)
 
-        self.best_solution = kwargs.pop('best_solution', None)
-        self.accuracy = kwargs.pop('accuracy', None)
-        self.niter = kwargs.pop('niter', None)
-        self.divergent = kwargs.pop('divergent', None)
-        self.slow_conv = kwargs.pop('slow_conv', None)
+        self.best_solution = best_solution
+        self.accuracy = accuracy
+        self.niter = niter
+        self.divergent = divergent
+        self.slow_conv = slow_conv
+
+        if kwargs:
```

Comment:
Is this needed? You don't do the same in `custom_model` above.

Reply:
Then I forgot to do it with `custom_model`. I really wanted to avoid accidental breakage due to this PR.